backend/operations_scanner.go

```diff
@@ -72,19 +72,19 @@ func (s *OperationsScanner) Run(logger *slog.Logger, stop <-chan struct{}) {
 	logger.Info("Polling Cluster Service every " + interval.String())
 	pollCSOperationsTicker := time.NewTicker(interval)
 
-	ctx := context.Background()
-
 	// Poll database immediately on startup.
-	s.pollDBOperations(ctx, logger)
+	s.pollDBOperations(logger)
 
-	for {
+	// The Context is cancelled when the process receives an interrupt signal.
+	// We let the periodic functions use their own Context so they can finish
+	// gracefully before we return.
+	for ctx.Err() == nil {
```

Comment:
IMO this is tricky code - just use `if ctx.Err() != nil { break }` in the `select` statement. Not obvious to me exactly the order of operations here - as written right now there are multiple steps:

1. `<-ctx.Done()`
2. `for ctx.Err() == nil`

If anyone ever changes this loop and execution is allowed to happen between the two, it will be harder to reason about the state of `ctx.Err()`.

Reply:
Label worked.  Updated 1st commit.